labgraph/examples/simple_viz_fixed_rate.py

```diff
@@ -0,0 +1,228 @@
+class PlotState(lg.State):
+    data: Optional[np.ndarray] = None
+
+
+# The configuration for the Plot
+class PlotConfig(lg.Config):
+    refresh_rate: float  # How frequently to refresh the bar graph
+    num_bars: int  # The number of bars to display (note this should be == num_features)
+
+
+# A node that creates a matplotlib bar graph that displays the produced data in
+# real-time
+class Plot(lg.Node):
+    INPUT = lg.Topic(RandomMessage)
+    state: PlotState
+    config: PlotConfig
+
+    def setup(self) -> None:
+        self.ax: Optional[matplotlib.axes.Axes] = None
+
+    # A subscriber method that simply receives data and updates the node's state
+    @lg.subscriber(INPUT)
+    def got_message(self, message: RandomMessage) -> None:
+        self.state.data = message.data
+
+    # A main method does not interact with topics, but has its own line of execution -
+    # this can be useful for Python libraries that must be run in the main thread. For
+    # example, scikit-learn and pyqtgraph are libraries that need the main thread.
+    @lg.main
+    def run_plot(self) -> None:
+        fig = plt.figure()
+        self.ax = fig.add_subplot(1, 1, 1)
+        self.ax.set_ylim((0, 1))
+        anim = animation.FuncAnimation(  # noqa: F841
+            fig, self._animate, interval=1 / self.config.refresh_rate * 1000
+        )
+        plt.show()
+        raise lg.NormalTermination()
+
+    def _animate(self, i: int) -> None:
+        if self.ax is None:
+            return
+        self.ax.clear()
+        self.ax.set_ylim([0, 1])
+        self.ax.bar(range(self.config.num_bars), self.state.data)
+
+
+# ======================================= DEMO =========================================
+
+
+# A graph for the demo in this example. Hooks together the AveragedNoise group
+# (containing NoiseGenerator and RollingAverager) and the Plot node.
+class Demo(lg.Graph):
+    AVERAGED_NOISE: AveragedNoise
+    PLOT: Plot
+
+    def setup(self) -> None:
+        # Provide configuration using global constants (but if we wanted to, we could
+        # have a configuration object provided to this graph as well).
+        self.AVERAGED_NOISE.configure(
+            AveragedNoiseConfig(
+                sample_rate=SAMPLE_RATE, num_features=NUM_FEATURES, window=WINDOW
+            )
+        )
+        self.PLOT.configure(
+            PlotConfig(refresh_rate=REFRESH_RATE, num_bars=NUM_FEATURES)
+        )
+
+    # Connect the AveragedNoise output to the Plot input
+    def connections(self) -> lg.Connections:
+        return ((self.AVERAGED_NOISE.OUTPUT, self.PLOT.INPUT),)
+
+    # Parallelization: Run AveragedNoise and Plot in separate processes
+    def process_modules(self) -> Tuple[lg.Module, ...]:
+        return (self.AVERAGED_NOISE, self.PLOT)
+
+    def logging(self) -> Dict[str, lg.Topic]:
+        return {"AVERAGED_NOISE.GENERATOR.OUTPUT": self.AVERAGED_NOISE.GENERATOR.OUTPUT}
+
```

Comment:
Do you want to write a description for this example, which would be useful for users to quickly understand the purpose of this example.